### PR TITLE
PDF Generation for Form 21-686C

### PR DESCRIPTION
**Pull request by kathleencrawford**
## Description of change

Adding the ability to generate a PDF for the form 21-686C.  This PR utilizes existing PDF generation code, while adding an implementation for the 21-686C.  The form 21-686C allows the user to add or remove dependents to their benefits.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#8759

## Things to know about this PR

There are no additional settings or logging added for this PR.  The PDF generation functionality already exists, this PR just adds form 21-686C to the existing functionality.

Regarding Code Climate issues...yes, there are a lot of lines of code in one file and lots of "similar blocks" of code.  This PDF is 15 pages long, it is a very complex form, that is why that one file is so big.  Most of it is just mapping the input fields from the PDF (there just happen to be so many of them on this form).  

## Testing

- [x] spec updated
- [x] local testing

### CODE REVIEW

File: lib/pdf_fill/forms/form_helper.rb [hunk 1]
```diff
@@ -43,6 +45,7 @@ def split_postal_code(address)
         return if address.blank?
 
         postal_code = address['postalCode']
+        postal_code = address['zip_code'] if postal_code.blank?
```
Comment (digitaldrk) [inline lib/pdf_fill/forms/form_helper.rb:48]: Does this work?

`postal_code = address['postalCode'] || address['zip_code']`

if so you could use that pattern in other areas like this :)

File: lib/pdf_fill/forms/va686c674.rb [hunk 2]
```diff
@@ -0,0 +1,1833 @@
...
+              'last' => {
+                key: 'form1[0].#subform[18].CHILDLastName[%iterator%]',
+                limit: 18,
+                question_num: 14,
+                question_suffix: 'C',
+                question_text: 'PREVIOUS MARRIAGE HISTORY >  PREVIOUS SPOUSE LAST NAME'
+              }
+              # 'suffix' => {
+              # @TODO
+              # } # end of suffix
```
Comment (digitaldrk) [inline lib/pdf_fill/forms/va686c674.rb:460]: Do you still need this?

File: lib/pdf_fill/forms/va686c674.rb [hunk 3]
```diff
@@ -0,0 +1,1833 @@
...
+              },
+              'last' => {
+                key: 'veteranSpouse.previousSpouse.lastName[%iterator%]',
+                limit: 18,
+                question_num: 15,
+                question_suffix: 'C',
+                question_text: 'PREVIOUS MARRIAGE HISTORY >  SPOUSES PREVIOUS SPOUSE LAST NAME'
+              },
+              'suffix' => {
+                # @TODO not in the form
```
Comment (digitaldrk) [inline lib/pdf_fill/forms/va686c674.rb:580]: Need the TODO?

File: lib/pdf_fill/forms/va686c674.rb [hunk 4]
```diff
@@ -0,0 +1,1833 @@
...
+            }, # end of start location
+            'reason_marriage_ended' => {
+              'death' => { key: 'spouse_marriage_history.reason_marriage_ended.death[%iterator%]' },
+              'divorce' => { key: 'spouse_marriage_history.reason_marriage_ended.divorce[%iterator%]' },
+              'annulment' => { key: 'spouse_marriage_history.reason_marriage_ended.annulment[%iterator%]' },
+              'other' => { key: 'spouse_marriage_history.reason_marriage_ended.other[%iterator%]' }
+            },
+            'reason_marriage_ended_other' => {
+              key: 'spouse_marriage_history.reason_marriage_ended_other[%iterator%]',
+              # limit: 12, # @TODO THIS BREAKS
```
Comment (digitaldrk) [inline lib/pdf_fill/forms/va686c674.rb:631]: Marking this one for visibility. Looks like it's not needed

File: lib/pdf_fill/forms/form_helper.rb (hunk 1 above)
Comment (kathleencrawford) [reply lib/pdf_fill/forms/form_helper.rb]: I've updated this as suggested - thanks!

File: lib/pdf_fill/forms/va686c674.rb (hunk 2 above)
Comment (kathleencrawford) [reply lib/pdf_fill/forms/va686c674.rb]: This is due to some differences between the front end and paper form, we still need to make a determination on how to handle these use cases.  I left the TODO comments in there as markers for when a decision is made and this is revisited.  I've added comments to hopefully make this a bit more clear.

File: lib/pdf_fill/forms/va686c674.rb (hunk 3 above)
Comment (kathleencrawford) [reply lib/pdf_fill/forms/va686c674.rb]: See above note about comments - thanks :)

File: lib/pdf_fill/forms/va686c674.rb (hunk 4 above)
Comment (kathleencrawford) [reply lib/pdf_fill/forms/va686c674.rb]: This has been removed, thanks!

**Pull request merged**